tools/cldr-code/src/main/java/org/unicode/cldr/util/SupplementalCalendarData.java

```diff
@@ -0,0 +1,229 @@
+package org.unicode.cldr.util;
+
+import com.ibm.icu.impl.locale.XCldrStub.ImmutableMap;
+import com.ibm.icu.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import org.unicode.cldr.icu.LDMLConstants;
+
+public class SupplementalCalendarData implements Iterable<String> {
+    /** an <era> element */
+    public static class EraData implements Comparable<EraData> {
+        static final int INDEX = 4; // index of our element
+
+        /** the <era> xpath */
+        private final XPathValue xpath;
+
+        EraData(final XPathValue xpath) {
+            this.xpath = xpath;
+            // TODO: validate (in unit test)
+        }
+
+        public int getType() {
+            return Integer.parseInt(xpath.getAttributeValue(INDEX, LDMLConstants.TYPE));
+        }
+
+        public String getStart() {
+            return xpath.getAttributeValue(INDEX, LDMLConstants.START);
+        }
+
+        public String getEnd() {
+            return xpath.getAttributeValue(INDEX, LDMLConstants.END);
+        }
+
+        public GregorianCalendar getStartCalendar() {
+            return forDateString(getStart());
+        }
+
+        public GregorianCalendar getEndCalendar() {
+            return forDateString(getEnd());
+        }
+
+        public String getCode() {
+            return xpath.getAttributeValue(INDEX, LDMLConstants.CODE);
+        }
+
+        public String[] getAliases() {
+            final String s = xpath.getAttributeValue(INDEX, LDMLConstants.ALIASES);
+            if (s == null) return "".split(" "); // empty array
+            return s.split(" ");
+        }
+
+        @Override
+        public String toString() {
+            return String.format(
+                    "ERA %d, [%s-%s] code=%s, aliases=%s",
+                    getType(), getStart(), getEnd(), getCode(), String.join(",", getAliases()));
+        }
+
+        private static final GregorianCalendar forDateString(String ymd) {
+            if (ymd == null) return null;
+            int multiplier = 1;
+            if (ymd.startsWith("-")) {
+                multiplier = -1;
+                ymd = ymd.substring(1);
+            }
+            final String[] parts = ymd.split("-");
+            try {
+                return new GregorianCalendar(
+                        multiplier * Integer.parseInt(parts[0]),
+                        Integer.parseInt(parts[1]) - 1,
+                        Integer.parseInt(parts[2]));
+            } catch (NumberFormatException nfe) {
+                throw new IllegalArgumentException("While parsing date string " + ymd, nfe);
+            }
+        }
+
+        private GregorianCalendar getLatest() {
+            GregorianCalendar l = getEndCalendar();
+            if (l == null) l = getStartCalendar();
+            return l;
+        }
+
+        /** only works within the same cal system */
```

Comment:
This is then too fragile; add a calendar ID to the EraData, then compare first by the calendarID, then by the era num ber.

Reply:
BTW, ComparisonChain is less error-prone.

Reply:
done